android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -280,12 +261,14 @@ private void setUpToolbar() {
     }
 
     private void setupEvent() {
-        event = realmRepo.getEventSync();
+        mainActivityViewModel.getEventSync().observe(this, eventData -> {
+            event = eventData;
 
-        if(event == null)
-            return;
+            if (event == null)
```

Comment:
Do this null check in the ViewModel, simply don't send anything if it is null. It will help keep the Activity as dumb as possible.

Reply:
this null check is present here so that setNavHeader() isn't called if the event is null.

Reply:
Yes, so what i am trying to say is, make sure you don't send null event from the ViewModel only. If you don't send a null event from the ViewModel, then on line 265, a non-null event will always be assigned and hence you won't need that null check.

Reply:
Even if I send non-null events from the ViewModel it's parameters can still be null. Hence, I don't think the null check should be removed from the activity

Reply:
If you don't send non null event from view model, the event parameter here can never be null. Can you give a use case and an example. It is possible to create entire Views like activity and fragments without a single if statement

Reply:
@iamareebjamal For null events I wrote ` if (event == null)
event = new Event();` in the viewModel
I get an error when for eg event.getSocialLinks() is used in the viewModel

Reply:
That's the thing, don't send the `new Event()` instance down the view and it will never be invalid

Reply:
@iamareebjamal okay I tried it. Now I don't get any errors in the viewModel but in the loadEvent function of the aboutFragment I get a NPE- text must not be null when event parameters  event.getEndsAt() and event.getSocialLinks() are called.

Can it be because nullability of the event is being checked in the fragment in other parts of the project ?

Reply:
No, that should not be the case, are you also checking if the event is valid before sending it to the view?

Reply:
I just tried it. Still the same error :cry: